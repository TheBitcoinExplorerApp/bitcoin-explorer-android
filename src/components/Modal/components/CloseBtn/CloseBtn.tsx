// eslint-disable-next-line import/no-extraneous-dependencies
import { TouchableOpacity, StyleSheet } from "react-native";
import CloseIcon from "../../../../assets/xIcon.svg";

type CloseBtnProps = {
  handleModalClose: () => void;
};

const styles = StyleSheet.create({
  closeButton: {
    width: 36,
    height: 36,
    justifyContent: "center",
    alignItems: "center",
    borderRadius: 100,
    padding: 5,
    backgroundColor: "#D9D9D9",
    position: "absolute",
    right: 0,
  },
});

export default function CloseBtn(props: CloseBtnProps) {
  const { handleModalClose } = props;

  return (
    <TouchableOpacity
      style={styles.closeButton}
      onPress={() => {
        handleModalClose();
      }}
      accessibilityLabel="Close Modal"
    >
      <CloseIcon width={20} height={20} />
    </TouchableOpacity>
  );
}
