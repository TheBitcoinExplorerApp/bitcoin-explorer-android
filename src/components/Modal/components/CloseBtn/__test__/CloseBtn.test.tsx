/* eslint-disable import/no-extraneous-dependencies */
import renderer from "react-test-renderer";
import CloseBtn from "../CloseBtn";

describe("CloseBtn", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<CloseBtn handleModalClose={() => {}} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
