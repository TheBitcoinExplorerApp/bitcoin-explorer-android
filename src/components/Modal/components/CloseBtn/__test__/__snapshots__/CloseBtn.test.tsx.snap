// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CloseBtn renders correctly 1`] = `
<View
  accessibilityLabel="Close Modal"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#D9D9D9",
      "borderRadius": 100,
      "height": 36,
      "justifyContent": "center",
      "opacity": 1,
      "padding": 5,
      "position": "absolute",
      "right": 0,
      "width": 36,
    }
  }
>
  <svg
    height={20}
    width={20}
  />
</View>
`;
